sky/serve/autoscalers.py

```diff
@@ -222,24 +265,41 @@ def _get_replica_ids_to_scale_down(num_limit: int) -> List[int]:
 
             status_order = serve_state.ReplicaStatus.scale_down_decision_order()
             launched_replica_infos_sorted = sorted(
-                launched_replica_infos,
+                launched_new_replica_infos,
                 key=lambda info: status_order.index(info.status)
                 if info.status in status_order else len(status_order))
 
             return [info.replica_id for info in launched_replica_infos_sorted
                    ][:num_limit]
 
-        if num_launched_replicas < self.target_num_replicas:
+        # Case 1. Once there is min_replicas number of
+        # ready new replicas, we will direct all traffic to them,
+        # we can scale down all old replicas.
+        if len(ready_new_replica_infos) >= self.min_replicas:
+            for info in ready_old_replica_infos:
+                all_replica_ids_to_scale_down.append(info.replica_id)
+
+        # Case 2. We have some not ready old replicas.
```

Comment:
I think it's fine since old replicas are going to be scaled down anyway. 
Do you mean just let these `NOT_READY` old replicas finish launching (and later they would be scaled-down)?

Reply:
Yeah, considering you have 1 NOT_READY old replica due to overflood and now you want to update to a new version; in this case, you might want to try to wait for the old replica to become ready (which will introduce several seconds downtime) rather than terminate it (which will introduce ~5min downtime)

Reply:
I see! This example makes a lot of sense.